### PR TITLE
Abstract shader semantics from SceneImporter

### PR DESCRIPTION
**Pull request by stevenvergenz**
Major changes:

* Abstracted shader-specific uniform names and keywords away from the SceneImporter and into the UniformMap classes. Enables swapping out the shader at a future date.
* Updated shaders so they're less broken. Texture packing and reflection probes work, vertex colors don't.
* Removed the shader input options, so users can no longer force the use of particular shaders. Instead, the included shaders are always used, albeit with the given LOD.

### CODE REVIEW

File: UnityGLTF/ASsets/UnityGLTF/Scripts/UniformMaps.meta [hunk 1]
```diff
@@ -1,7 +1,7 @@
 fileFormatVersion: 2
-guid: b8468e126fb62374b8675d3125eb2eec
+guid: e766804828a966240ab9d312c8abb342
```
Comment (MSblgross) [inline UnityGLTF/ASsets/UnityGLTF/Scripts/UniformMaps.meta]: why did the meta's change?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 2]
```diff
@@ -407,203 +400,115 @@ protected virtual MaterialCacheData CreateMaterial(GLTF.Schema.Material def, int
...
 
 					if (pbr.MetallicRoughnessTexture != null)
 					{
 						var texture = pbr.MetallicRoughnessTexture.Index.Value;
-						material.SetTexture("_MetallicRoughnessMap", CreateTexture(texture));
+						mrMapper.MetallicRoughnessTexture = CreateTexture(texture);
+						mrMapper.MetallicRoughnessTexCoord = pbr.MetallicRoughnessTexture.TexCoord;
 
-						ApplyTextureTransform(pbr.MetallicRoughnessTexture, material, "_MetallicRoughnessMap");
+						//ApplyTextureTransform(pbr.MetallicRoughnessTexture, material, "_MetallicRoughnessMap");
```
Comment (MSblgross) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: ?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 3]
```diff
@@ -407,203 +400,115 @@ protected virtual MaterialCacheData CreateMaterial(GLTF.Schema.Material def, int
...
-					{
-						material.EnableKeyword("OCC_METAL_ROUGH_ON");
-					}
-					else
-					{
-						material.SetTexture("_OcclusionMap", CreateTexture(texture.Value));
-
-						ApplyTextureTransform(def.OcclusionTexture, material, "_OcclusionMap");
-					}
+					//ApplyTextureTransform(def.OcclusionTexture, material, "_OcclusionMap");
```
Comment (MSblgross) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: ?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 4]
```diff
@@ -407,203 +400,115 @@ protected virtual MaterialCacheData CreateMaterial(GLTF.Schema.Material def, int
...
 					var texture = def.EmissiveTexture.Index.Value;
-					material.EnableKeyword("EMISSION_MAP_ON");
-					material.EnableKeyword("_EMISSION");
-					material.SetTexture("_EmissionMap", CreateTexture(texture));
-					material.SetInt("_EmissionUV", def.EmissiveTexture.TexCoord);
+					mapper.EmissiveTexture = CreateTexture(texture);
+					mapper.EmissiveTexCoord = def.EmissiveTexture.TexCoord;
 
-					ApplyTextureTransform(def.EmissiveTexture, material, "_EmissionMap");
+					//ApplyTextureTransform(def.EmissiveTexture, material, "_EmissionMap");
```
Comment (MSblgross) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: ?

File: UnityGLTF/ASsets/UnityGLTF/Scripts/UniformMaps.meta (hunk 1 above)
Comment (stevenvergenz) [reply UnityGLTF/ASsets/UnityGLTF/Scripts/UniformMaps.meta]: I'm more curious why there is an old version at all, tbh. I make that folder in this branch.

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs (hunk 2 above)
Comment (stevenvergenz) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: I tentatively removed support for that extension until we can figure out where we should put it. I'm invested in getting that right, its disabling is only temporary.

**Pull request merged**